x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java

```diff
@@ -408,7 +412,7 @@ public Builder(TrainedModelConfig config) {
             this.definition = config.definition == null ? null : new LazyModelDefinition(config.definition);
             this.description = config.getDescription();
             this.tags = config.getTags();
-            this.metadata = config.getMetadata();
+            this.metadata = config.getMetadata() == null ? null : new HashMap<>(config.getMetadata());
```

Comment:
The package private ctor also wraps this map in line 151

```this.metadata = metadata == null ? null : Collections.unmodifiableMap(metadata);```

Make the ctor argument a `UnmodifiableMap` and `getMetadata()` return an `UnmodifiableMap` and I think that simplifies these maps being wrapped in a map in a map etc